pallets/external-validators-rewards/src/lib.rs

```diff
@@ -254,62 +264,70 @@ pub mod pallet {
 
     impl<T: Config> tp_traits::OnEraEnd for Pallet<T> {
         fn on_era_end(era_index: EraIndex) {
-            if let Some(utils) = Self::generate_era_rewards_utils(era_index, None) {
-                let tokens_inflated = T::EraInflationProvider::get();
-
-                let ethereum_sovereign_account = T::RewardsEthereumSovereignAccount::get();
-                if let Err(err) =
-                    T::Currency::mint_into(&ethereum_sovereign_account, tokens_inflated.into())
-                {
-                    log::error!(target: "ext_validators_rewards", "Failed to mint inflation into Ethereum Soverein Account: {err:?}");
-                    log::error!(target: "ext_validators_rewards", "Not sending message since there are no rewards to distribute");
-                    return;
-                }
+            let token_location = T::TokenLocationReanchored::get();
+            let token_id = T::TokenIdFromLocation::convert_back(&token_location);
+
+            if let Some(token_id) = token_id {
+                if let Some(utils) = Self::generate_era_rewards_utils(era_index, None) {
+                    let tokens_inflated = T::EraInflationProvider::get();
+
+                    let ethereum_sovereign_account = T::RewardsEthereumSovereignAccount::get();
+                    if let Err(err) =
```

Comment:
Non blocking question: What are the scenario in which this can fail?

Reply:
mm I guess if you try to mint an amount that is less than the ED for insance